server/core/src/https/views/profile.rs

```diff
@@ -52,13 +136,210 @@ pub(crate) async fn view_profile_get(
+        .await?;
+
+    let time = time::OffsetDateTime::now_utc() + time::Duration::new(60, 0);
+    let can_rw = uat.purpose_readwrite_active(time);
+    // TODO: A bit overkill to request scimEffectiveAccess here.
+    let (scim_person, _) = crate::https::views::admin::persons::get_person_info(
+        uat.uuid,
+        state,
+        &kopid,
+        client_auth_info.clone(),
+    )
+    .await?;
+
+    let primary_index = query
+        .emails_indexes
+        .iter()
+        .position(|ei| ei == &query.primary_email_index)
+        .unwrap_or(0);
+    let new_mails = query
+        .emails
+        .iter()
+        .enumerate()
+        .map(|(ei, email)| ScimMail {
+            primary: ei == primary_index,
+            value: email.to_string(),
+        })
+        .collect();
+    let old_primary_mail = scim_person
+        .mails
+        .iter()
+        .find(|sm| sm.primary)
+        .map(|sm| sm.value.clone());
+
+    let emails_are_same = scim_person.mails == new_mails;
+
+    let profile_view = ProfileChangesPartialView {
+        menu_active_item: ProfileMenuItems::UserProfile,
+        can_rw,
+        person: scim_person,
+        primary_mail: old_primary_mail,
+        new_attrs: ProfileAttributes {
+            account_name: query.account_name,
+            display_name: query.display_name,
+            emails: new_mails,
+        },
+        emails_are_same,
+        new_primary_mail: query.emails.get(primary_index).cloned(),
+    };
+
+    Ok((
+        HxPushUrl(Uri::from_static("/ui/profile/diff")),
+        profile_view,
+    )
+        .into_response())
+}
+
+pub(crate) async fn view_profile_diff_confirm_save_post(
+    State(state): State<ServerState>,
+    Extension(kopid): Extension<KOpId>,
+    VerifiedClientInformation(client_auth_info): VerifiedClientInformation,
+    DomainInfo(domain_info): DomainInfo,
+    // Form must be the last parameter because it consumes the request body
+    Form(query): Form<CommitSaveProfileQuery>,
+) -> axum::response::Result<Response> {
+    let uat: UserAuthToken = state
+        .qe_r_ref
+        .handle_whoami_uat(client_auth_info.clone(), kopid.eventid)
+        .map_err(|op_err| HtmxError::new(&kopid, op_err, domain_info.clone()))
+        .await?;
+
+    let filter = filter_all!(f_and!([f_id(uat.uuid.to_string().as_str())]));
+
+    if let Some(account_name) = query.account_name {
+        state
+            .qe_w_ref
+            .handle_setattribute(
+                client_auth_info.clone(),
+                uat.uuid.to_string(),
+                ATTR_NAME.to_string(),
+                vec![account_name],
+                filter.clone(),
+                kopid.eventid,
+            )
+            .map_err(|op_err| HtmxError::new(&kopid, op_err, domain_info.clone()))
+            .await?;
+    }
+    if let Some(display_name) = query.display_name {
+        state
+            .qe_w_ref
+            .handle_setattribute(
```

Comment:
This should be done in a single modification, not 5 different ones.

Reply:
Ahhh https://github.com/kanidm/kanidm/pull/3535 needs me to finish so you can do this in a single scim post.

Reply:
I saw the pr got merged but I'm failing to figure out what scim function is relevant for this bulk operation ?

Reply:
You want https://github.com/kanidm/kanidm/blob/master/server/lib/src/server/scim.rs#L110